apps/labs/posts/libsf-error-state.md

```diff
@@ -0,0 +1,744 @@
+
+```C++
+// SCIPY_DLL
+// inspired by https://github.com/abseil/abseil-cpp/blob/20240116.2/absl/base/config.h#L736-L753
+//
+// When building sf_error_state as a DLL, this macro expands to `__declspec(dllexport)`
+// so we can annotate symbols appropriately as being exported. When used in
+// headers consuming a DLL, this macro expands to `__declspec(dllimport)` so
+// that consumers know the symbol is defined inside the DLL. In all other cases,
+// the macro expands to nothing.
+// Note: SCIPY_DLL_{EX,IM}PORTS are set in scipy/special/meson.build
+#if defined(SCIPY_DLL_EXPORTS)
+    #define SCIPY_DLL __declspec(dllexport)
+#elif defined(SCIPY_DLL_IMPORTS)
+    #define SCIPY_DLL __declspec(dllimport)
+#else
+    #define SCIPY_DLL
+#endif
+```
+
+As soon as I had a chance, I fired up a Windows VM again and put together [a PR](https://github.com/scipy/scipy/pull/20937)
+implementing Axel's solution. After a couple misteps, MSVC builds were working again. There would be no need to push
+back the release date. A couple weeks later, fellow Quansight Labs member and LFortran/LPython core developer Gagandeep
+Singh (czgdp1807) submitted [a PR](https://github.com/scipy/scipy/pull/20985) to add an MSVC CI job, plugging the gap in
+SciPy's coverage.
+
+### Thread safety
+
+In the introduction I'd mentioned that CPython's global interpreter lock (GIL) makes it easier to extend Python with C
+or other compiled languages since one doesn't need to worry about the thread safety of wrapped code. Still, being able
+to have only one thread in a running Python process execute Python code at a time is greatly limiting for multicore
+applications. In October 2023, a Python Enhancement Proposal (PEP) was accepted to make the GIL optional. This proposal,
+[PEP 703](https://peps.python.org/pep-0703/), is well worth reading for its in-depth summary of the surrounding
+issues. CPython 3.13 launched with an optional
+[free-threaded mode](https://docs.python.org/3.13/whatsnew/3.13.html#free-threaded-cpython)
+which supports running with the GIL disabled. Free-threading in Python offers great promise, but to take advantage of
+it, extensions need to be rewritten with thread safety in mind.
+
+`libsf_error_state` is obviously not thread safe. There is a global array carrying the current state of special
+function error handling policies with nothing to stop competing threads from trying to access the same memory
+location. Simultaneous modifications could even leave an entry in some corrupt and indeterminate state. This situation
+is known as a data race and leads to undefined behavior in C and C++ []. Weird things can happen when there is a data race.
+
+The latest tale in the saga of `libsf_error_state` is [a PR](https://github.com/scipy/scipy/pull/21956) from Edgar
+Margffoy (andfoy)—a member of Quansight Labs' free-threaded Python team—to ensure thread safety by declaring the array
```

Comment:
github handle? and spaces around the dashes

Reply:
I've seen inconsistent recommendations for spaces around em-dashes, e.g. https://writing.stackexchange.com/questions/8555/em-dash-space-or-no-space-between-words. Grepping for "—" among the labs blog posts, I found a pretty equal distribution of spaces vs no spaces. If you like spaces better, I'll do that.